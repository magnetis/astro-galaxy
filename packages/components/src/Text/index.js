import styled from 'styled-components';
import { color, space, border, layout, buttonStyle, typography } from 'styled-system';
import { font } from '@magnetis/astro-galaxy-core';

const Text = styled.p`
  color: ${props => props.theme.colors[props.color]};
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  ${font};
  font-size: ${props => props.theme.fontSizes.text[props.fontSize]};
`;

Text.displayName = 'Text';

Text.defaultProps = {
  font: 'primary',
  color: 'moon900',
  fontSize: 'medium',
};

Text.propTypes = {
  ...color.propTypes,
  ...space.propTypes,
  ...border.propTypes,
  ...layout.propTypes,
  ...buttonStyle.propTypes,
  ...typography.propTypes,
};

export default Text;
