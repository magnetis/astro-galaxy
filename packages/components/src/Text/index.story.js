import React from 'react';
import { storiesOf } from '@storybook/react';
import Text from './index.js';

storiesOf('Text', module)
  .add('primary', () => <Text>Astro Galaxy</Text>)
  .add('secondary', () => <Text font="secondary">Astro Galaxy</Text>)
  .add('colors', () => (
    <>
      <Text color="earth400">earth400</Text>
      <Text color="mars400">mars400</Text>
      <Text color="moon400">moon400</Text>
      <Text color="space400">space400</Text>
      <Text color="sun400">sun400</Text>
      <Text color="uranus400">uranus400</Text>
      <Text color="venus400">venus400</Text>
    </>
  ))
  .add('sizes', () => (
    <>
      <Text fontSize="verySmall">verySmall</Text>
      <Text fontSize="small">small</Text>
      <Text fontSize="medium">medium</Text>
      <Text fontSize="large">large</Text>
    </>
  ))
  .add('as span', () => <Text as="span">medium</Text>);
