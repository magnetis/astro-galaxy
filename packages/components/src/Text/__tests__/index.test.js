import React from 'react';
import { theme } from '@magnetis/astro-galaxy-core';

import Text from '../index';

describe('Text', () => {
  it('renders correctly', () => {
    const json = rendererCreateWithTheme(<Text>Text</Text>).toJSON();
    expect(json).toMatchSnapshot();
  });

  describe('when `font` prop is `secondary`', () => {
    it('renders with font-family Lato', () => {
      const json = rendererCreateWithTheme(<Text font="secondary">Text</Text>).toJSON();
      expect(json).toHaveStyleRule('font', 'normal normal 400 normal 16px/1.5 Lato,sans-serif');
    });
  });

  describe('when `color` prop is set', () => {
    it('renders with a defined color', () => {
      const json = rendererCreateWithTheme(<Text color="earth400">Text</Text>).toJSON();
      expect(json).toHaveStyleRule('color', theme.colors.earth400);
    });
  });

  describe('when `fontSize` prop is set', () => {
    it('renders with a defined font size', () => {
      const json = rendererCreateWithTheme(<Text fontSize="verySmall">Text</Text>).toJSON();
      expect(json).toHaveStyleRule('font-size', theme.fontSizes.texts.verySmall);
    });
  });
});
