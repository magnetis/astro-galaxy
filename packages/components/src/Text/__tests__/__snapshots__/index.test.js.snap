// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #32383c;
  font-size: medium;
  font: normal normal 600 normal 16px/1.5 Poppins,sans-serif;
  font-size: 16px;
}

<p
  className="c0"
  color="moon900"
  fontSize="medium"
>
  Text
</p>
`;
